slack-api-client/src/main/java/com/slack/api/rate_limits/metrics/impl/BaseMemoryMetricsDatastore.java

```diff
@@ -70,6 +77,16 @@ public RequestStats getStats(String executorName, String teamId) {
         return stats;
     }
 
+    @Override
+    public ExecutorServiceProvider getExecutorServiceProvider() {
+        return this.executorServiceProvider;
+    }
+
+    @Override
+    public void setExecutorServiceProvider(ExecutorServiceProvider executorServiceProvider) {
+        this.executorServiceProvider = executorServiceProvider;
```

Comment:
TODO: if we provide this setter method, `this.cleanerExecutor ` needs to be recreated when this method is called.

Reply:
done